lib/datasource/gitlab-packages/index.ts

```diff
@@ -0,0 +1,88 @@
+import { cache } from '../../util/cache/package/decorator';
+import { GitlabHttp } from '../../util/http/gitlab';
+import { joinUrlParts } from '../../util/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { datasource } from './common';
+import type { GitlabPackage } from './types';
+
+// Gitlab Packages API: https://docs.gitlab.com/ee/api/packages.html
+
+export class GitlabPackagesDatasource extends Datasource {
+  static readonly id = datasource;
+
+  protected override http: GitlabHttp;
+
+  override caching = true;
+
+  override customRegistrySupport = true;
+
+  override defaultVersioning = 'docker';
```

Comment:
@rarkins I'm not happy with this, we don't do this on github-releases. So i would suggest to remove this. WDYT?

Reply:
Are packages Docker images or generic? You suggest keeping semver default?

Reply:
i think they are generic, as gitlab supports different registry types

Reply:
yes, keeping semver and force the user to change if required.

Reply:
Ok